src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -399,11 +398,13 @@ void WgGeometryData::appendStroke(const WgPolyline* polyline, const RenderStroke
                 } else if (stroke->join == StrokeJoin::Miter) {
                     WgPoint nrm = (nrm0 + nrm1).normal();
                     float cosine = nrm.dot(nrm0);
-                    if ((cosine != 0.0f) && (abs(cosine) != 1.0f) && (abs(wdt / cosine) <= stroke->miterlimit * 2)) {
-                        appendRect(v1 + nrm * (wdt / cosine), v1 + offset0, v1 + offset1, v1 - nrm * (wdt / cosine));
+                    float mitterRation = 1.0f / std::sin((std::acos(dir0.dot(dir1.negative()))/2.0f));
+                    if ((cosine != 0.0f) && (abs(cosine) < 1.0f) && (mitterRation <= stroke->miterlimit)) {
```

Comment:
cosine != 0 -> mathZero

since vectors used to calculate cosine are normalized, can cosine be > 1?

Reply:
It can be equal to 1.0 and its not a case for us